R/readKane.R

```diff
@@ -0,0 +1,63 @@
+#' Raw Read for data corresponding to the Kane dataset
+#'
+#' Download and read in the raw data tables from Kane, E., Ping, C., Chapin, S. and 
+#' Valentine, D. with Bonanza LTER. doi:10.6073/pasta/d09433eee2cb6587eca672864cf7e90f
+#' 
+#' This is not post-processed or QA/QCed by this package.
+#'
+#' @param dataDir filename for download directory
+#' @param download boolean that will download the files from repository 
+#'
+#' @return a list with meta data and raw table reads. This will be 49 KB in size.
+#'
+#' @export
+
+dataDir <- '~/Desktop/Research/Kane' #edit to reflect personal directory
+
+
+#data read in for citation [2] not included
+  #url confirmation is needed, see ##TODO from 114_Kane.Rmd
+#KaneBiophysicalData is missing header information that could not be found online, contacting of researchers is necessary
+
+
+readKane2004 <- function(dataDir, download = TRUE, verbose = FALSE){
+ 
+  
+  url_table <- data.frame(File_Base = c('KaneSoilData.txt', 'KaneBiophysicalData.txt', 'KaneMetaData.xml'),
+                          Download_URL = c('http://www.lter.uaf.edu/php/download_data.php?f=/data_files/ascii/files/190_2031_all_soil_profile_depths_carbon_BD.txt', 'http://www.lter.uaf.edu/php/download_data.php?f=/data_files/ascii/files/190_1608_sitesummarydata.txt', 'http://www.lter.uaf.edu/eml/BNZeml132.xml'))
+  
+  
+  #for loop to read in data from url_table if it does not exist in local repository
+  for(ii in 1:nrow(url_table)){
+    DataFile <- file.path(dataDir, url_table$File_Base[ii])
+    if(!(file.exists(DataFile))){
+      download.file(url_table$Download_URL[ii], destfile= DataFile, quiet=FALSE)
+    }
+  }
+  
+  # #this code was not used as it requires additional package dependency
+  # plyr::d_ply(
+  #   url_table,
+  #   c('Download_URL','File_Base'),
+  #   function(xx){
+  #     DataFile <- file.path(dataDir, xx$File_Base)
+  #     if(!(file.exists(DataFile))){
+  #     download.file(xx$Download_URL, destfile= DataFile, quiet=FALSE)
+  #     }
+  #     }
+  #   )
+  
+  #reading in data for R
+  readKanePitData <- read_csv(file.path(dataDir, 'KaneSoilData.txt'))
+  readKaneBiophysicalData <- read_csv(file.path(dataDir, 'KaneBiophysicalData.txt'), col_names = c("Site Description", "Site ID", "X3", "X4", "X5", "X6", "X7", "X8", "X9", "X10", "X11", "X12", "X13", "X14")) #note that most headers are missing
+
+  #ans and its return
+  ans <- list(downloadFiles = c(file.path(dataDir,'KaneSoilData', 'KaneBiophysicalData', 'KaneMetaData')),
```

Comment:
Try to only recreate the filenames once and then save it to a variable. Repeated code like this is more prone to error. In this case you forgot the file extensions on these download files!